utils.py

```diff
@@ -0,0 +1,112 @@
+from syslog import LOG_DEBUG
+
+import iso8601
+from pyparsing import Word, Suppress, nums, Optional, Regex, pyparsing_common, alphanums
+
+
+def parse_log_lines(body):
+    parser = Parser()
+    return [l for l in parser.parse_rfc6587(body) if filter_log_lines(l)]
+
+
+def filter_log_lines(msg):
+    """Return true if we should send to kinesis."""
+    sev = msg["severity"]  # e.g. LOG_DEBUG
+    body = msg["message"]
+
+    if sev >= LOG_DEBUG:
+        return False
+
+    if body.startswith("DEBUG   "):
+        return False
+
+    return True
+
+
+class Parser(object):
+    def __init__(self):
```

Comment:
I'm not sure on this part of the code,

What would be ideal is we just simply split the octet framed individual syslog messages out, and return them as a list and send them that way without parsing them. So, basically if the block contains 3 messages, we'd end up parsing that into 3 strings and just submitting those as is; they'd be converted into a list as though they came in individually.

We can then just use hindsight's syslog decoders to do any additional parsing that is required and it will be consistent across the board with other syslog imports and normalized the same way.

Reply:
One other thought here too -- would the intent be that there would be a single API gateway for multiple drains? In the scenario would there be a way to tell which Heroku app sent the log message?

If not should we run a gateway/lambda per application category or something, or does the log message contain something we could use to identify where it came from?

Reply:
This sounds great. I would love to remove this section of the code. The one question I have though is we had initially discussed filtering out messages within the Lambda func, should we not do this? If we do still want to filter out some messages then I imagine the func will need to parse out the contents of the syslog log line.

>If not should we run a gateway/lambda per application category or something, or does the log message contain something we could use to identify where it came from?

In looking at the logs, sometimes they do sometimes they do not. I think an easy way to accomplish this is a set of credentials per app+env and we can use the username as the identifier.